frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -83,7 +83,7 @@ type DashboardProps = {
   dashboardBeforeEditing: IDashboard | null;
   isEditingParameter: boolean;
   isDirty: boolean;
-  dashboard: IDashboard;
+  dashboard: IDashboard | null;
```

Comment:
The value passed from `DashboardApp` is actually `Dashboard | null` so we have to account for that

Reply:
The other option is to put a `if (!dashboard)` check on `DashboardApp`

Reply:
If `DashboardApp` is the only consumer of this component, then yes I think it makes sense to leave the responsibility to the caller, and make `dashboard` prop here non-nullable. Otherwise either approach is fine.

Reply:
Actually,  I think we need to keep the null checks here. The `Dashboard` component handles the initialization, loading, and error states for the actual dashboard components while `DashboardApp` handles toasts and notifications. The prop for getting the dashboard is actually a selector within the Dashboard